components/prism-sas.js

```diff
@@ -237,7 +237,22 @@
 			lookbehind: true,
 			inside: args
 		},
-
+		'macro-string-functions': {
+			pattern: /((?:^|\s|=))%(?:NRBQUOTE|NRQUOTE|NRSTR|BQUOTE|QUOTE|STR)\(.*?(?:[^%]\))/i,
+			lookbehind: true,
+			inside: {
+				function: {
+					pattern: /%(?:NRBQUOTE|NRQUOTE|NRSTR|BQUOTE|QUOTE|STR)/i,
+					alias: 'keyword'
+				},
+				string: {
+					pattern: /(\()(?:[\s\S]*?)(?=(?!%)\))/,
```

Comment:
1. `(?!%)\)` We know that the next character is `)`, so `(?!%)` will always be true.

The pattern is now `(\()[\s\S]*?(?=\))`. This has two other problems: 1) It can potentially produce empty tokens, so let's make it `(\()[\s\S]+?(?=\))`. 2) `[\s\S]+?(?=\))` is slower than `[^)]+`.

```suggestion
					pattern: /(\()[^)]+/,
```